--- a/mani_skill/examples/motionplanning/base_motionplanner/motionplanner.py
+++ b/mani_skill/examples/motionplanning/base_motionplanner/motionplanner.py
@@ -63,7 +63,7 @@
             user_joint_names=joint_names,
             move_group=move_group,
         )
-        planner.set_base_pose(np.hstack([self.base_pose.p, self.base_pose.q]))
+        planner.set_base_pose(mplib.pymp.Pose(p=self.base_pose.p, q=self.base_pose.q))
         planner.joint_vel_limits = np.asarray(planner.joint_vel_limits) * self.joint_vel_limits
         planner.joint_acc_limits = np.asarray(planner.joint_acc_limits) * self.joint_acc_limits
         return planner
@@ -149,17 +149,15 @@
         self._update_grasp_visual(pose)
         pose = self._transform_pose_for_planning(pose)
         result = self.planner.plan_screw(
-            np.concatenate([pose.p, pose.q]),
+            mplib.pymp.Pose(p=pose.p, q=pose.q),
             self.robot.get_qpos().cpu().numpy()[0],
             time_step=self.base_env.control_timestep,
-            use_point_cloud=self.use_point_cloud,
         )
         if result["status"] != "Success":
             result = self.planner.plan_screw(
-                np.concatenate([pose.p, pose.q]),
+                mplib.pymp.Pose(p=pose.p, q=pose.q),
                 self.robot.get_qpos().cpu().numpy()[0],
                 time_step=self.base_env.control_timestep,
-                use_point_cloud=self.use_point_cloud,
             )
             if result["status"] != "Success":
                 print(result["status"])
